src/components/FieldInput/index.tsx

```diff
@@ -15,7 +15,7 @@ import { RiEyeCloseLine, RiEyeLine } from 'react-icons/ri';
 import { FormGroup, FormGroupProps } from '@/components/FormGroup';
 
 export interface FieldInputProps
-  extends FieldProps,
+  extends FieldProps<string | number | readonly string[] | undefined>,
```

Comment:
On all FieldXX components, you should remove the `useState` and `useEffect` for `isTouched` data management and use the stuff provided by `useField` hook

Reply:
Same for the use of `required` prop, you can get the `isRequired` value return by useField instead

Reply:
Same for `showError` that you can replace by the useField provided value `shouldDisplayError`